Userland/Libraries/LibC/stdio.cpp

```diff
@@ -31,7 +31,8 @@ struct FILE {
         : m_fd(fd)
         , m_mode(mode)
     {
-        __pthread_mutex_init(&m_mutex, nullptr);
+        __pthread_mutexattr_t mutex_attr { __PTHREAD_MUTEX_RECURSIVE };
```

Comment:
What's the reason for making the mutex recursive?

Reply:
POSIX [says](https://pubs.opengroup.org/onlinepubs/9699919799/functions/ftrylockfile.html) that `FILE` should maintain a lock count. So if this is to be implemented in a mutex, it should be recursive.